include/api.php

```diff
@@ -6223,13 +6223,10 @@ function api_friendica_profile_show($type)
 
 			// select all users from contact table, loop and prepare standard return for user data
 			$users = [];
-			$nurls = Contact::select(['id', 'nurl'], ['uid' => api_user(), 'profile-id' => $rr['id']]);
-
-			if (DBA::isResult($nurls)) {
-				foreach ($nurls as $nurl) {
-					$user = api_get_user($a, $nurl['nurl']);
-					($type == "xml") ? $users[$k++ . ":user"] = $user : $users[] = $user;
-				}
+			$nurls = Contact::selectToArray(['id', 'nurl'], ['uid' => api_user(), 'profile-id' => $rr['id']]);
+			foreach ($nurls as $nurl) {
```

Comment:
The function ```Contact::selectToArray``` can return a boolean value when an SQL error occurred.

Reply:
Then it's a bad function and I will change it immediately.

Reply:
Alright, fixed. I found a couple more issues with Item and Attach.

Reply:
The boolean value indicates an error - or ```true``` when applied to some SQL query that doesn't return a dataset.

Reply:
Which isn't the case for `toArray` which expects a dataset.

Reply:
You could apply this function to a database command like "UPDATE" - then it would return a "true" as boolean.

Reply:
But it doesn't make sense functionally, there's no array to be extracted from an UPDATE query statement, so the empty array would be functionally correct.

Now, if you check the boolean state of the return value of `DBA::toArray` on an UPDATE statement to guess if the update was successful, you're doing all kinds of wrongs that have nothing to do with `DBA::toArray` anymore.

Reply:
I assumed that this function had been used in the legacy ```q(...)``` function - where we don't know if this is a "SELECT" or an "UPDATE". But luckily we do check for boolean return values before that function call.